crates/ruma-common/src/events/room/create.rs

```diff
@@ -62,6 +80,30 @@ impl RoomCreateEventContent {
     }
 }
 
+impl RedactContent for RoomCreateEventContent {
+    type Redacted = RedactedRoomCreateEventContent;
+
+    fn redact(self, version: &RoomVersionId) -> Self::Redacted {
+        match version {
+            RoomVersionId::V1
+            | RoomVersionId::V2
+            | RoomVersionId::V3
+            | RoomVersionId::V4
+            | RoomVersionId::V5
+            | RoomVersionId::V6
+            | RoomVersionId::V7
+            | RoomVersionId::V8
+            | RoomVersionId::V9
+            | RoomVersionId::V10 => {
+                let mut redacted = Self::new_v11();
+                redacted.creator = self.creator;
+                redacted
+            }
```

Comment:
We can now implement `Default` for `RoomCreateEventContent`, then this can be

```suggestion
            | RoomVersionId::V10 => Self {
                creator: self.creator,
                ..Self::default()
            }
```

Reply:
I'm not sure I like having the `Default` implementation as part of the public API. If we set all the fields to their default value, we have a bastard event that uses the defaults of room v1, with `room_version` set to v1 but doesn't have a creator which is invalid for room v1.

We could have a `Default` implementation that defaults to `new_v11`, with `room_version` set to v11.

Reply:
You're right, probably not a good idea to implement that then 😅

Reply:
Oh but we can still use struct update syntax w/ `..Self::new_v11()`.

Reply:
Except we also need to set the room version. I.e. the current impl is actually wrong, isn't it?

Reply:
Yes, because I forgot to update the room version in `new_v11()` before. And now I forgot to update this part to match that change.

Reply:
Now it should be good.